src/blas/impl/KokkosBlas2_gemv_impl.hpp

```diff
@@ -569,7 +566,12 @@ struct TwoLevelGEMV {
       IndexType yrow = team.league_rank() * 32 + i;
       if(yrow < (IndexType) A_.extent(0))
       {
-        y_(yrow) = beta_ * y_(yrow) + alpha_ * blockResult[i];
+        Scalar betaTerm;
+        if(beta_ == KAT::zero())
+          betaTerm = KAT::zero();
```

Comment:
Looking at how this is handled in the the rest of the file, why not go with
```c++
if(beta_ == KAT::zero())
  y_(yrow) = alpha_ * blockResult[I];
else
  y_(yrow) = beta_ * y_(yrow) + alpha_ blockResult[I];
```
Also I did not read the whole, file so maybe this is a trivial question: why are we not assuming that `yrow` and `row` are the same?

Reply:
``row`` and ``yrow`` are different because first all the threads in the team compute the Aij*xj term for i = ``row``, and then only 32 threads in a TeamThreadRange write the results back from shared to ``yrow``.

Reply:
I don't know why I wrote the beta zero case differently here, I just made it consistent.